### PR TITLE
Report

### PR DESCRIPTION
**Pull request by VovaDrobko**

### CODE REVIEW

**Review by hrafov — DISMISSED** [review]

File: .idea/.name [hunk 1]
```diff
@@ -0,0 +1 @@
+InternalProject
```
Comment (old-zoomer) [inline .idea/.name]: Remove .idea/ files from commit

File: InternalProject.iml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (old-zoomer) [inline InternalProject.iml]: remove .iml files

File: allure-results/0753d57a-7638-4862-9ad3-dcdcf76d6f50-result.json [hunk 3]
```diff
@@ -0,0 +1 @@
...rker.run(SuiteRunnerWorker.java:86)\r\n\tat org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)\r\n\tat org.testng.TestNG.runSuitesLocally(TestNG.java:1226)\r\n\tat org.testng.TestNG.runSuites(TestNG.java:1144)\r\n\tat org.testng.TestNG.run(TestNG.java:1115)\r\n\tat org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)\r\n\tat org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)\r\n\tat org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)\r\n\tat org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)\r\n\tat org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)\r\n\tat org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)\r\n"},"stage":"finished","start":1536630640376,"stop":1536630640443,"attachments":[{"name":"{0}","source":"90b3bf3e-b048-48e2-831a-ca556ce633b0-attachment","type":"text/plane"}],"parameters":[{"name":"iTestResult","value":"[TestResult name=logInTest status=FAILURE method=LogInTest.logInTest()[pri:0, instance:com.epam.lab.LogInTest@4816c290] output={null}]"}]}],"parameters":[]}
```
Comment (old-zoomer) [inline allure-results/0753d57a-7638-4862-9ad3-dcdcf76d6f50-result.json]: remove allure-results/

File: target/classes/testng.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
```
Comment (old-zoomer) [inline target/classes/testng.xml]: remove target from PR

File: src/main/java/com/epam/lab/page/LoginPagePo.java [hunk 5]
```diff
@@ -12,14 +13,17 @@
     private PageElement submitButton;
 
 
+    @Step("PageElementPO Step: Enter login: {0} in login input field in method: {enterLogin}...")
```
Comment (old-zoomer) [inline src/main/java/com/epam/lab/page/LoginPagePo.java]: this methods should be in BO layer, PO layer should just return web elements

**Review by KnazevValeriy — DISMISSED** [review]

**Review by old-zoomer — CHANGES_REQUESTED** [review]

File: .idea/.name (hunk 1 above)
Comment (VovaDrobko) [reply .idea/.name]: I have already removed .idea files from branch

File: InternalProject.iml (hunk 2 above)
Comment (VovaDrobko) [reply InternalProject.iml]: I have alredy removed .iml files from branch

File: allure-results/0753d57a-7638-4862-9ad3-dcdcf76d6f50-result.json (hunk 3 above)
Comment (VovaDrobko) [reply allure-results/0753d57a-7638-4862-9ad3-dcdcf76d6f50-result.json]: I have alredy removed allure-result files from branch

File: target/classes/testng.xml (hunk 4 above)
Comment (VovaDrobko) [reply target/classes/testng.xml]: I have alredy removed target files from branch

File: src/main/java/com/epam/lab/page/LoginPagePo.java (hunk 5 above)
Comment (VovaDrobko) [reply src/main/java/com/epam/lab/page/LoginPagePo.java]: I have done all required modifications for LoginPagePO / LoginPageBO

**Review by MysiukR — DISMISSED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,6 @@
-.idea/
```
Comment (old-zoomer) [inline .gitignore:1]: do not change .gitignore in this pull request

File: README.md [hunk 7]
```diff
@@ -1 +1,16 @@
-# TestProject
```
Comment (old-zoomer) [inline README.md]: do not change readme in this pull request

File: src/main/java/com/epam/lab/business/LoginPageBo.java [hunk 8]
```diff
@@ -14,10 +15,25 @@ public LoginPageBo() {
...
-        loginPagePo.enterLogin(login);
-        loginPagePo.enterPassword(password);
-        loginPagePo.clickSubmit();
+        enterLogin(login);
+        enterPassword(password);
+        clickSubmit();
+    }
+
+    @Step("PageElementBO Step: Enter login: {0} in login input field in method: {enterLogin}...")
+    private void enterLogin(String login) {
```
Comment (old-zoomer) [inline src/main/java/com/epam/lab/business/LoginPageBo.java]: Make these 3 methods public

**Review by old-zoomer — CHANGES_REQUESTED** [review]

**Review by hrafov — DISMISSED** [review]

File: README.md (hunk 7 above)
Comment (VovaDrobko) [reply README.md]: I only have changed version of allure. It was wrong.

File: src/main/java/com/epam/lab/business/LoginPageBo.java (hunk 8 above)
Comment (VovaDrobko) [reply src/main/java/com/epam/lab/business/LoginPageBo.java]: I have made them public.

File: .gitignore [hunk 9]
```diff
@@ -1,4 +1,6 @@
...
-target/
-allure-results/
+.idea/*
 *.iml
+target/*
+allure-results/*
+README
```
Comment (old-zoomer) [inline .gitignore]: README should not be ignored

File: src/main/java/business/LoginPageBo.java [hunk 10]
```diff
@@ -1,26 +0,0 @@
-package business;
```
Comment (old-zoomer) [inline src/main/java/business/LoginPageBo.java]: Don't change duplicate files in this pull request

**Review by old-zoomer — CHANGES_REQUESTED** [review]

File: src/main/java/business/LoginPageBo.java (hunk 10 above)
Comment (VovaDrobko) [reply src/main/java/business/LoginPageBo.java]: I have reverted dublicates

File: .gitignore (hunk 9 above)
Comment (VovaDrobko) [reply .gitignore]: I have fixed this issue.

**Review by KnazevValeriy — APPROVED** [review]

**Review by hrafov — APPROVED** [review]

**Review by hrafov — APPROVED** [review]

**Review by MysiukR — APPROVED** [review]

**Review by old-zoomer — APPROVED** [review]

**Pull request merged**